src/test/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBeanTest.java

```diff
@@ -278,6 +280,39 @@ public void testBeanConvertersUri3() {
         }
     }
 
+    @Test
+    public void testBeanConverterPatternArray() throws Exception {
+        final ConverterBean bean = new ConverterBean();
+        final DefaultConfiguration config = new DefaultConfiguration("bean");
+        final String patternString = "^a*$,^b*$, ^c*$";
+        final List<String> expectedPatternStrings = Arrays.asList("^a*$", "^b*$", " ^c*$");
+        config.addProperty("patterns", patternString);
+        bean.configure(config);
+
+        final List<String> actualPatternStrings = Arrays.stream(bean.patterns)
+                .map(Pattern::pattern)
+                .collect(Collectors.toUnmodifiableList());
+
+        assertWithMessage("invalid size of result")
+                .that(bean.patterns)
+                .hasLength(3);
+        assertWithMessage("invalid result")
+                .that(actualPatternStrings)
+                .containsExactlyElementsIn(expectedPatternStrings);
+    }
+
+    @Test
+    public void testBeanConverterPatternArraySingleElement() throws Exception {
+        final ConverterBean bean = new ConverterBean();
+        final DefaultConfiguration config = new DefaultConfiguration("bean");
+        config.addProperty("patterns", "");
+        bean.configure(config);
```

Comment:
testBeanConverterPatternArraySingleElement

Empty is not a single, please make 2 tests, for single pattern and separate for empty.

Reply:
done